caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -36,6 +36,16 @@ func init() {
 	caddyconfig.RegisterAdapter("caddyfile", caddyfile.Adapter{ServerType: ServerType{}})
 }
 
+// GloballyConfiguredApp represents the configuration for a
+// Caddy app module which was parsed from a global options block.
+type GloballyConfiguredApp struct {
```

Comment:
Since all apps are "globally configured" (i.e. there's at most one per config), I suggest a rename to simplify things:

```suggestion
type App struct {
```

Or something like that. I think it'll be easy to find this in the godocs too.

Reply:
Might also be worth mentioning in the comment that this is for non-standard apps only; standard apps (http, tls, pki) would be overwritten.